storage/fs.go

```
@@ -70,20 +70,27 @@ func (st *FSStorage) filePath(hash string) string {
 func (st *FSStorage) Store(hash string, value []byte) error {
 	fileDir := path.Join(st.rootPath, "files", hash[0:2], hash[2:4])
 	filePath := path.Join(fileDir, hash)
+	fileTemp := fmt.Sprintf("%s_%s", filePath, utils.RandString(16))
 
 	err := os.MkdirAll(fileDir, os.ModePerm)
 	if err != nil {
 		return err
 	}
 
+	err = ioutil.WriteFile(fileTemp, value, 0644)
+	if err != nil {
+		return err
+	}
+
 	unlock, err := utils.Lock(filePath)
```

We seem to be ignoring this error here. Maybe the empty file bug originated from that? 🤔

---

Nice catch!